fish-rust/src/builtins/set.rs

```diff
@@ -1,5 +1,498 @@
+use crate::{
+    env::{EnvMode, EnvVar, Environment},
+    wutil::wcstoi::wcstoi_partial,
+};
+
 use super::prelude::*;
+use crate::wutil::wcstoi;
+use once_cell::sync::Lazy;
 
-pub fn set(parser: &Parser, streams: &mut IoStreams<'_>, args: &mut [WString]) -> Option<c_int> {
+// PORTING: I don't know if `_` is evaluated at runtime or compile-time and whether translations are/will be embedded/loaded at runtime
+// FIXME: (once localization works in Rust) These should separate `%ls: ` and the trailing `\n`, like in builtins/string
+const MISMATCHED_ARGS: Lazy<&wstr> =
+    Lazy::new(|| wgettext_str(L!("%ls: given %d indexes but %d values\n")));
```

Comment:
`_` calls `gettext` at runtime, just like `wgettext!`.
I have been adding a wgettext_fmt call to wherever the constant is used. Not ideal but reasonably simple